populator-machinery/controller.go

```diff
@@ -751,191 +708,241 @@ func (c *controller) syncPvc(ctx context.Context, key, pvcNamespace, pvcName str
 		Unstructured: unstructured,
 		Recorder:     c.recorder,
 	}
-	if "" == pvc.Spec.VolumeName {
 
-		// Ensure the PVC has a finalizer on it so we can clean up the stuff we create
-		err = c.ensureFinalizer(ctx, pvc, c.pvcFinalizer, true)
-		if err != nil {
-			return err
+	// *** Here is the first place we start to create/modify objects ***
+	// Check if the original PVC is getting deleted, if so stop volume populating
+	// and clean up the temporary resources created by the volume populator
+	if pvc.DeletionTimestamp == nil && pvc.Status.Phase != "Terminating" {
```

Comment:
nil check for pvc.Status?

Reply:
Same as above, pvc.Status is not a pointer, we don't need to do nil check